soco/music_services/accounts.py

```diff
@@ -0,0 +1,181 @@
+            self.nickname,
+            hex(id(self))
+        )
+
+    def __str__(self):
+        return self.__repr__()
+
+    @staticmethod
+    def _get_account_xml(soco):
+        """Fetch the account data from a Sonos device.
+
+        Args:
+            soco (SoCo): a SoCo instance to query. If soco is none, a
+                random device will be used.
+
+        Returns:
+            (str): a byte string containing the account data xml
+
+        """
+        # It is likely that the same information is available over UPnP as well
+        # via a call to
+        # systemProperties.GetStringX([('VariableName','R_SvcAccounts')]))
+        # This returns an encrypted string, and, so far, we cannot decrypt it
+        device = soco or SoCo.any_soco()
+        log.debug("Fetching account data from %s", device)
+        settings_url = "http://{0}:1400/status/accounts".format(
+            device.ip_address)
+        result = requests.get(settings_url).content
+        log.debug("Account data: %s", result)
+        return result
+
+    @classmethod
+    def get_accounts(cls, soco=None):
+        """Get a dict containing all accounts known to the Sonos system.
+
+        Args:
+            soco (SoCo, optional): a SoCo instance to query. If None, a random
+            instance is used. Defaults to None
+
+        Returns:
+            dict: A dict containing account instances. Each key is the
+            account's serial number, and each value is the related Account
+            instance. Accounts which have been marked as deleted are excluded.
+
+        Note:
+            Although an Account for TuneIn is always present, it is handled
+            specially by Sonos, and will not appear in the returned dict. Any
+            existing Account instance will have its attributes updated to
+            those currently stored on the Sonos system.
+
+        """
+
+        root = XML.fromstring(cls._get_account_xml(soco))
+        # <ZPSupportInfo type="User">
+        #   <Accounts
+        #   LastUpdateDevice="RINCON_000XXXXXXXX400"
+        #   Version="8" NextSerialNum="5">
+        #     <Account Type="2311" SerialNum="1">
+        #         <UN>12345678</UN>
+        #         <MD>1</MD>
+        #         <NN></NN>
+        #         <OADevID></OADevID>
+        #         <Key></Key>
+        #     </Account>
+        #     <Account Type="41735" SerialNum="3" Deleted="1">
+        #         <UN></UN>
+        #         <MD>1</MD>
+        #         <NN>Nickname</NN>
+        #         <OADevID></OADevID>
+        #         <Key></Key>
+        #     </Account>
+        # ...
+        #   <Accounts />
+
+        # pylint: disable=protected-access
+
+        accounts = root.findall('.//Account')
+        result = {}
+        for account in accounts:
+            serial_number = account.get('SerialNum')
+            is_deleted = True if account.get('Deleted') == '1' else False
+            if cls._all_accounts.get(serial_number):
```

Comment:
Took me a few read-throughs to figure out lines 142-153, maybe a short comment.

Maybe also write out account as the variable name